### PR TITLE
[NPU] Add Gemma4 Sliding Window Attention support on Ascend backend

### PR DESCRIPTION
**Pull request by syy-hw**
## Motivation

Gemma4 uses a hybrid Sliding Window Attention (SWA) architecture where 5/6 of transformer layers use local (windowed) attention and 1/6 use global attention. The Ascend NPU backend lacked SWA support, preventing Gemma4 models from running correctly on Ascend NPU.

### SWA Attention Mask & Graph Mode Decode (ascend_backend.py)
  - Add `get_swa_mask()` to `AscendAttnMaskBuilder` — generates dynamic sliding window mask for decode.
  - Add `swa_mask` field to `ForwardMetadata` for graph capture/replay.
  - Pre-allocate `swa_mask` and `swa_indices` buffers during graph capture to avoid per-replay allocation.
  - Update mask content during `init_forward_metadata_replay_cuda_graph` based on current seq lengths and sliding window size.

### SWA Paged Block Table Construction (ascend_backend.py)
  - Add helper methods: `_is_swa_layer()`, `_can_use_tnd()`, `_get_swa_page_aligned_starts()`, `_get_swa_paged_seq_lens_cpu_int/list()`, `_build_swa_paged_block_tables()`,  `_get_paged_attention_inputs()`.
  - Dynamically build SWA-aware page-aligned block tables with correct window clipping for FIA/SDPA/`PA` paths.

### SDPA Fallback SWA Support (ascend_torch_native_backend.py)
- Add `sliding_window_size` and `full_to_swa_mapping` parameters to SDPA forward functions for SWA window clipping in the fallback path.

### NPU Memory Pool SWA Parameters (memory_pool_npu.py)
- Pass `swa_head_num`, `swa_head_dim`, `swa_v_head_dim` parameters to parent `MHATokenToKVPool` for SWA KV pool allocation.

### Sliding Window Size Conversion (gemma4_causal.py)
- Use `get_attention_sliding_window_size(config)` for SWA layers and set `-1` for full attention layers, aligning with sglang's exclusive window convention.

### NPU Graph Compatibility Fix (gemma4_mm.py)
- Replace boolean indexing (`ple_ids[mask] = pad_id`) with `torch.where` for NPU graph mode compatibility.

### Attention Backend Selection (server_args.py)
- Update Gemma4 attention backend selection logic to accept `ascend` backend on NPU.

Evaluated with [evalscope](https://github.com/modelscope/evalscope).

| Model | Dataset | Ascend NPU | Baseline |
|-------|---------|------------|---------------|
| Gemma4-E2B-it | AIME 2026 | 46.67%| 37.5% |
| Gemma4-E2B-it | MMMU Pro | 55.55%| 44.2% |
| Gemma4-E4B-it | AIME 2026 | 46.67%| 42.5%|
| Gemma4-E4B-it | MMMU Pro | 60.92%| 52.6%|
| Gemma4-31B-it | AIME 2026 | 93.33%| 89.2%|
| Gemma4-31B-it | MMMU Pro | 78.67%| 76.9%|
| Gemma4-26B-A4B-it | AIME 2026 | 86.67%| 88.3%|
| Gemma4-26B-A4B-it | MMMU Pro | 76.18%| 73.8%|

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #26266324354](https://github.com/sgl-project/sglang/actions/runs/26266324354)
Latest PR Test (Extra): :x: [Run #26266324350](https://github.com/sgl-project/sglang/actions/runs/26266324350)

### CODE REVIEW

File: python/sglang/srt/models/gemma4_causal.py [hunk 1]
```diff
@@ -222,7 +222,9 @@ def __init__(
 
         layer_type = config.layer_types[layer_id]
         self.sliding_window = (
-            config.sliding_window if layer_type == "sliding_attention" else None
+            get_attention_sliding_window_size(config)
+            if layer_type == "sliding_attention"
+            else -1
```
Comment (kpham-sgl) [inline python/sglang/srt/models/gemma4_causal.py:283]: Nice catch!
